eclair-core/src/main/scala/fr/acinq/eclair/wire/CommonCodecs.scala

```diff
@@ -52,9 +52,14 @@ object CommonCodecs {
   // this codec can be safely used for values < 2^63 and will fail otherwise
   // (for something smarter see https://github.com/yzernik/bitcoin-scodec/blob/master/src/main/scala/io/github/yzernik/bitcoinscodec/structures/UInt64.scala)
   val uint64overflow: Codec[Long] = int64.narrow(l => if (l >= 0) Attempt.Successful(l) else Attempt.failure(Err(s"overflow for value $l")), l => l)
-
   val uint64: Codec[UInt64] = bytes(8).xmap(b => UInt64(b), a => a.toByteVector.padLeft(8))
 
+  val uint64overflowSat: Codec[Satoshi] = uint64overflow.xmapc(l => Satoshi(l))(_.toLong)
+  val uint64overflowMSat: Codec[MilliSatoshi] = uint64overflow.xmapc(l => MilliSatoshi(l))(_.toLong)
+
+  val uint64Sat: Codec[Satoshi] = uint64.xmapc(l => Satoshi(l.toBigInt.longValue()))(sat => UInt64(sat.toLong))
```

Comment:
Do we even need this one? It seems like `Long` is definitely enough for all amounts, even for amounts in msat a `Long` should be able to contain all the Bitcoins ever created...
So we should only use the `overflow` version because a value that overflows is likely not a valid amount, right?

Reply:
Yes Long should be enough: we have 21*1e6 btc, which is 21*1e6*1e8 = 21*1e14 sat, which is 21*1e17 msat which we can still represent with a Long.
And I'd rather have the codecs named satoshi and millisatoshi ?